code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm

```diff
@@ -63,14 +63,23 @@
 	disconnect_from_area()
 	assign_to_area()
 
-/obj/machinery/atmospherics/components/unary/vent_scrubber/proc/assign_to_area()
-	var/area/area = get_area(src)
+/obj/machinery/atmospherics/components/unary/vent_scrubber/on_enter_area(datum/source, area/area_to_register)
+	assign_to_area(area_to_register)
+	. = ..()
+
+/obj/machinery/atmospherics/components/unary/vent_scrubber/proc/assign_to_area(area/target_area = get_area(src))
+	var/area/area = target_area
 	area?.air_scrubbers += src
```

Comment:
This shouldn't silently pass and should stack_trace if the passed in area is null